### PR TITLE
Add default timeout values for gitlab/github api clients

### PR DESCRIPTION
**Pull request by lechuk47**
APPSRE-3888

### CODE REVIEW

**Review by rporres — COMMENTED** [review]
Looks great, I love the tests

Just nit-picking: can we make the timeouts a float, e.g. `0.1` so that they fail faster?

File: reconcile/test/test_utils_github_api.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+from unittest.mock import patch
+from requests.exceptions import ConnectTimeout
+from reconcile.utils.github_api import GithubApi
+
+
+class TestGithubApi(TestCase):
+
+    @patch("reconcile.utils.github_api.GH_BASE_URL", "http://198.18.0.1")
+    def test_github_client_timeout(self):
+        with patch('reconcile.utils.github_api.SecretReader', autospec=True) \
```
Comment (Piojo) [inline reconcile/test/test_utils_github_api.py:11]: I tend to dislike using this `with` format, any changes to indentation will make your tests use the real class and it'll be very confusing. Why don't you decorate this method with the new base URL and the `SecretReader`?

File: reconcile/test/test_utils_gitlab_api.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from unittest import TestCase
+from unittest.mock import patch
+from requests.exceptions import ConnectTimeout
+from reconcile.utils.gitlab_api import GitLabApi
+
+
+class TestGitlabApi(TestCase):
+
+    def test_gitlab_client_timeout(self):
+        with patch('reconcile.utils.gitlab_api.SecretReader', autospec=True) \
```
Comment (Piojo) [inline reconcile/test/test_utils_gitlab_api.py:10]: As above.

**Review by Piojo — CHANGES_REQUESTED** [review]
Trivial requests, but they can cause problems in the near future.

File: reconcile/test/test_utils_gitlab_api.py (hunk 2 above)
Comment (lechuk47) [reply reconcile/test/test_utils_gitlab_api.py]: answered above.

File: reconcile/test/test_utils_github_api.py (hunk 1 above)
Comment (lechuk47) [reply reconcile/test/test_utils_github_api.py]: I think this concern applies in a more general python level, or any other languages that manage the code structures with indents (like nested loops, etc). 
Changed to a decorator flavour.

**Comment by lechuk47** [comment]
> Looks great, I love the tests
> 
> Just nit-picking: can we make the timeouts a float, e.g. `0.1` so that they fail faster?

@rporres Changed in the gitlab test, github requires a integer so `1` is the minimum.

**Review by Piojo — APPROVED** [review]
This is awesome. Thanks.

**Comment by rporres** [comment]
thx for taking a look @lechuk47 . This looks great, indeed

**Pull request merged**